layer1/transaction/backend.go

```diff
@@ -89,13 +89,13 @@ func newReplacedInfo(newTxn *types.Transaction, originalTxInfo info) info {
 	)
 }
 
-// Internal struct to keep track of transactions retries groups
+// group is an internal struct to keep track of transactions retries groups
 type group struct {
 	InternalGroup   []common.Hash  `json:"internalGroup"` // slice where we keep track of all tx in a group
 	receiptResponse *SharedReceipt `json:"-"`             // struct used to send/share the receipt
 }
 
-// creates a new group
+// newGroup creates a new group
```

Comment:
Comment shouldn't just restate the name.

```suggestion
// newGroup for receipt responses.
```